### PR TITLE
FOGL-5872.patch : backup table sorting by date and added file drag/drop feature 

### PR DESCRIPTION
**Pull request by mshariq-nerd**

### CODE REVIEW

**Review by monika-Dianomic — DISMISSED** [review]

File: src/app/components/common/file-upload-modal/file-upload-modal.component.html [hunk 1]
```diff
@@ -6,15 +6,16 @@
...
+      (dragover)="onDragOver($event)" (drop)="onDropSuccess($event)">
         <label class="file-label">
           <input class="file-input" type="file" (change)="attachBackupFile($event)">
           <span class="file-cta">
             <span class="file-icon">
               <i class="fas fa-upload"></i>
             </span>
             <span class="file-label">
-              Choose a file…
+              Browse/Drag & drop file
```
Comment (praveen-garg) [inline src/app/components/common/file-upload-modal/file-upload-modal.component.html:18]: Browse/Drag & drop the backup tarball

File: src/app/components/common/file-upload-modal/file-upload-modal.component.ts [hunk 2]
```diff
@@ -33,17 +33,33 @@ export class FileUploadModalComponent implements OnInit {
...
+  }
+
+  // Read file
+  onFileChange(files: File[]) {
+    if (files.length !== 0) {
+      const fileName = files[0].name;
       this.fileName = fileName;
-      if (event.target.files.length > 0) {
-        const file = event.target.files[0];
+      if (files.length > 0) {
```
Comment (praveen-garg) [inline src/app/components/common/file-upload-modal/file-upload-modal.component.ts:45]: are we allowing more than one file?

File: src/app/components/core/backup-restore/backup-restore.component.html [hunk 3]
```diff
@@ -52,7 +52,13 @@
...
               <tr>
-                <th>Date &amp; Time</th>
+                <th>
+                  <div id="sort-order" (click)="sort()">
+                    Date &amp; Time
+                    <i *ngIf="ascSort" #asc class="fas fa-caret-up" aria-hidden="true"></i>
+                    <i *ngIf="!ascSort" #desc class="fas fa-caret-down" aria-hidden="true"></i>
+                  </div>
+                </th>
                 <th>Status</th>
```
Comment (praveen-garg) [inline src/app/components/core/backup-restore/backup-restore.component.html:62]: also allow to sort on status, i.e. group by status and newer will remain on top

File: src/app/components/core/backup-restore/backup-restore.component.html [hunk 4]
```diff
@@ -52,7 +52,13 @@
           <table class="table is-hoverable is-narrow scroll is-responsive" cellpadding="0" cellspacing="0">
             <thead>
               <tr>
-                <th>Date &amp; Time</th>
+                <th>
+                  <div id="sort-order" (click)="sort()">
+                    Date &amp; Time
+                    <i *ngIf="ascSort" #asc class="fas fa-caret-up" aria-hidden="true"></i>
```
Comment (praveen-garg) [inline src/app/components/core/backup-restore/backup-restore.component.html:58]: the current selected must be disabled.

File: src/app/components/common/file-upload-modal/file-upload-modal.component.ts (hunk 2 above)
Comment (mshariq-nerd) [reply src/app/components/common/file-upload-modal/file-upload-modal.component.ts:45]: No, one file only.

**Review by praveen-garg — DISMISSED** [review]

**Review by praveen-garg — APPROVED** [review]
lgtm!

**Pull request merged**